packages/engine/src/behave-graph/nodes/Profiles/Engine/helper/stateHelper.ts

```diff
@@ -125,3 +130,88 @@ export function getStateGetters() {
   }
   return getters
 }
+
+type State = {
+  systemUUID: SystemUUID
+}
+
+const initialState = (): State => ({
+  systemUUID: '' as SystemUUID
+})
+
+export function getStateListeners() {
+  const getters: NodeDefinition[] = []
+  const skipped: string[] = []
+  for (const [stateName, state] of Object.entries(Engine.instance.store.stateMap)) {
+    if (skipState.includes(stateName)) {
+      skipped.push(stateName)
+      continue
+    }
+    const outputsockets = generateStateNodeSchema(state)
+    if (Object.keys(outputsockets).length === 0) {
+      skipped.push(stateName)
+      continue
+    }
+    const node = makeEventNodeDefinition({
+      typeName: `engine/component/on ${stateName} change`,
```

Comment:
Do we have to make a separate event node for each component? Can we do a single `engine/onComponentChange` node?

Reply:
i suppose, but then we have to dynamically change the handles each time, i suppose its possible now, ill do it

Reply:
got now a single node will require too much hardcoded elements